plutus-ledger/src/Ledger/Tx.hs

```diff
@@ -96,16 +101,17 @@ type PrivateKey = Crypto.XPrv
 -- This datatype was created in order to be used in
 -- 'Ledger.Constraints.processConstraint', specifically with the constraints
 -- 'MustSpendPubKeyOutput' and 'MustSpendScriptOutput'.
---
--- TODO Add 'Either DatumHash Datum' field for 'PublicKeyChainIndexTxOut'.
 data ChainIndexTxOut =
-    PublicKeyChainIndexTxOut { _ciTxOutAddress :: Address
-                             , _ciTxOutValue   :: Value
+    PublicKeyChainIndexTxOut { _ciTxOutAddress         :: Address
+                             , _ciTxOutValue           :: V1.Value
+                             , _ciTxOutPublicKeyDatum  :: V2.OutputDatum
+                             , _ciTxOutReferenceScript :: Maybe V1.Script
                              }
-  | ScriptChainIndexTxOut { _ciTxOutAddress   :: Address
-                          , _ciTxOutValidator :: Either ValidatorHash Validator
-                          , _ciTxOutDatum     :: Either DatumHash Datum
-                          , _ciTxOutValue     :: Value
+  | ScriptChainIndexTxOut { _ciTxOutAddress         :: Address
+                          , _ciTxOutValue           :: V1.Value
+                          , _ciTxOutScriptDatum     :: Either V1.DatumHash V1.Datum
+                          , _ciTxOutReferenceScript :: Maybe V1.Script
+                          , _ciTxOutValidator       :: Either V1.ValidatorHash V1.Validator
```

Comment:
I think it would be helpful to add a comment explaining the difference between the two. I got confused as well :)